test/gomega/matchers/have_protocol_test.go

```diff
@@ -0,0 +1,28 @@
+package matchers_test
+
+import (
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/gloo/test/gomega/matchers"
+)
+
+var _ = Describe("HaveProtocol", func() {
+
+	It("ignores if there is no protocol specified", func() {
+		httpResponse := &http.Response{
+			Proto: matchers.HTTP1Protocol,
+		}
+		Expect(httpResponse).To(matchers.HaveProtocol(""))
```

Comment:
I'm not quite sure how this test works. As I read it, we define a response with the http1 protocol, and then assert there is an empty protocol. Can you help me understand what I'm missing?

Reply:
This test verifies that if an empty protocol is passed to the method, it always succeeds